python/trace-python.c

```diff
@@ -889,30 +860,32 @@ static struct uftrace_python_symbol *convert_function_addr(PyObject *frame, PyOb
 		Py_INCREF(code);
 	}
 
+	if (is_pyfunc)
+		func_name = get_python_funcname(frame, code, &is_main);
+	else
+		func_name = get_c_funcname(frame, code);
+
+	if (func_name == NULL)
+		return NULL;
+
+	/* code is not used anymore */
+	Py_DECREF(code);
+
 	while (*p) {
 		parent = *p;
 		iter = rb_entry(parent, struct uftrace_python_symbol, node);
 
-		/* just compare pointers of the code object */
-		if (iter->code == code) {
-			Py_DECREF(code);
+		/* compare func_name */
+		if (iter->name == func_name) {
```

Comment:
I don't think we can do this.  It should call `strcmp()`.

Reply:
fixed